src/main/java/com/google/sps/servlets/EventFreeTimeServlet.java

```diff
@@ -0,0 +1,198 @@
+    Long duration = eventStorageObject.getEvent(eventId).getDuration();
+    if (timeRange.isDateTimeSet()) {
+      response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+      return;
+    }
+
+    DateTime startDateTime = new DateTime(timeRange.convertStartTimeToString());
+    DateTime startDateTimeWithShift = new DateTime(startDateTime.getValue() + timeRange.getShift());
+    DateTime endDateTime = new DateTime(timeRange.convertEndTimeToString());
+    DateTime endDateTimeWithShift = new DateTime(endDateTime.getValue() + timeRange.getShift());
+    BatchRequest batchRequest = new BatchRequest(flow.getTransport(), flow.getRequestInitializer())
+        .setBatchUrl(new GenericUrl("https://www.googleapis.com/batch/calendar/v3"));
+    String ownerId = eventStorageObject.getEvent(eventId).getOwnerID();
+
+    if (userService.getCurrentUser() == null || !userService.getCurrentUser().getUserId().equals(ownerId)) {
+      response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+      return;
+    }
+
+    List<String> joinedParticipantsIds = eventStorageObject.getEvent(eventId).getJoinedIDs();
+    if (!joinedParticipantsIds.contains(ownerId)) {
+      joinedParticipantsIds.add(ownerId);
+    }
+
+    List<Credential> usersCredentials;
+    try {
+      usersCredentials = getCredentialsFromUserList(joinedParticipantsIds);
+    } catch (InterruptedException e2) {
+      e2.printStackTrace();
+      response.setStatus(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+      return;
+    }
+
+    usersCredentials.forEach(userCredential -> {
+      try {
+        createAndAddBatchRequest(batchRequest, callback, userCredential, startDateTimeWithShift, endDateTimeWithShift);
+        batchRequest.execute();
+        returnFreeTimesFromInterval(startDateTimeWithShift.getValue(), endDateTimeWithShift.getValue(), duration, busyTimesForAttendees, response);
+        return;
+      } catch (IOException e1) {
+        // TODO Auto-generated catch block
+        e1.printStackTrace();
+      } catch (GeneralSecurityException e1) {
+        // TODO Auto-generated catch block
+        e1.printStackTrace();
+      }
+      return;
+    });
+
+  }
+
+  private void createAndAddBatchRequest(BatchRequest batchRequest, JsonBatchCallback<FreeBusyResponse> callback,
+      Credential userCredential, DateTime start, DateTime end) throws IOException, GeneralSecurityException {
+    Calendar userCalendar = loadCalendarClient(userCredential);
+    FreeBusyRequest fBusyRequest = new FreeBusyRequest().setTimeMin(start).setTimeMax(end)
+        .setItems(Collections.singletonList(new FreeBusyRequestItem().setId("primary")));
+    HttpRequest request = userCalendar.freebusy().query(fBusyRequest).buildHttpRequest();
+    batchRequest.queue(request, FreeBusyResponse.class, GoogleJsonErrorContainer.class, callback);
+  }
+
+  private void returnFreeTimesFromInterval(Long start, Long end, Long duration, Set<Time> busyTimes, HttpServletResponse response)
```

Comment:
Name said returns but return type is void. Either really return it or make it clear that it writes it to the response.